features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/value/IpValue.java

```diff
@@ -67,26 +71,143 @@ private void parse(final StringValue input) {
                 }
                 // Ensure each range is an ip address
                 for (StringValue rangedValue : rangedValues) {
-                    verifyIpAddress(rangedValue);
+                    if (rangedValue.contains("/")) {
+                        throw new IllegalArgumentException("Ranged value may not contain a CIDR expression");
+                    }
                 }
-                // Verify the range itself
-                final IPAddressRange range = new IPAddressRange(rangedValues.get(0).getValue(), rangedValues.get(1).getValue());
-                ranges.add(range);
+                ranges.add(new IpAddressRange(rangedValues.get(0), rangedValues.get(1)));
             } else {
-                verifyIpAddress(eachValue);
-                ranges.add(new IPAddressRange(eachValue.getValue(), eachValue.getValue()));
+                ranges.add(new IpAddressRange(eachValue));
             }
         }
     }
 
     public boolean isInRange(final String address) {
-        return ranges.stream().anyMatch(r -> r.contains(address));
+        return ranges.stream().anyMatch(r -> r.isInRange(address));
     }
 
-    private static void verifyIpAddress(final StringValue stringValue) {
-        Objects.requireNonNull(stringValue);
-        if (!InetAddresses.isInetAddress(stringValue.getValue())) {
-            throw new IllegalArgumentException("Provided ip address '" + stringValue.getValue() + "' is invalid");
+    private static class IpAddressRange {
```

Comment:
Is this class even needed? As `IpAddressMatcher` already dispatches, it can be included directly in the parent class. And it would solve the `IpAddressRange` vs `IPAddressRange` conflict.